### PR TITLE
Client branch merge

### PR DESCRIPTION
**Pull request by shingchan95**
This merge incorporates several enhancements from the client branch, including:

- Updated README.md for clarity.
- Added a requirements.txt file.
- Improved code readability by adding PEP8 standard docstrings to functions.
- Implemented error handling in relevant functions for improved robustness.

These changes collectively enhance the overall quality and maintainability of the codebase.

### CODE REVIEW

File: src/client/main.py [hunk 1]
```diff
@@ -5,6 +5,22 @@
 
 
 def send_data(data):
```
Comment (husky960) [inline src/client/main.py:7]: Looks great! As we talked, we should switch to using a config file for the server details.

File: src/client/utils/serialise.py [hunk 2]
```diff
@@ -2,8 +2,30 @@
 
 
 class Serialiser:
```
Comment (husky960) [inline src/client/utils/serialise.py:4]: Looks good in general.

- We need to import pickle? 
- It might be good to implement error handling.

**Review by husky960 — APPROVED** [review]
Looks great! It is great to see some error handling!
There's a few minor comments.

File: src/client/main.py (hunk 1 above)
Comment (shingchan95) [reply src/client/main.py:7]: thank you, and yes thats a good idea

File: src/client/utils/serialise.py (hunk 2 above)
Comment (shingchan95) [reply src/client/utils/serialise.py:4]: yes we do need pickle for serialise and deserialise

**Pull request merged**